netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -105,9 +114,39 @@ public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exc
       }
 
       private void fail(ChannelHandlerContext ctx, Throwable exception) {
-        log.log(Level.FINEST, "TLS negotiation failed for new client.", exception);
+        log.log(Level.FINE, errorMessage(ctx), exception);
         ctx.close();
       }
+
+      private String errorMessage(ChannelHandlerContext ctx) {
+        StringBuilder builder = new StringBuilder("TLS negotiation failed for new client. ");
+        SSLEngine engine = sslHandler(ctx).engine();
+        if (engine instanceof OpenSslEngine) {
+          builder.append("OpenSSL version: ");
+          builder.append("0x").append(Integer.toHexString(OpenSsl.version()));
```

Comment:
Wasn't the OpenSsl version was pretty useless, since it doesn't reflect the actual version installed?